File: Android App/app/src/main/java/revert/shredsheets/models/themes/ShredsheetsTheme.java
package revert.shredsheets.models.themes;

import android.graphics.Paint;

public abstract class ShredsheetsTheme {
    public ShredsheetsTheme() {
    }

    public static ShredsheetsTheme[] getThemes() {
        return new ShredsheetsTheme[] { new DefaultTheme(), new BlueTheme() };
    }

    public boolean[] defaultVector = new boolean[]{true, false, true, false, true, false, true, false, false, false, false, false};

    public abstract boolean[] getDegreeHighlightingVector();
    public abstract void setDegreeHighlightingVector(boolean[] values);

    public abstract Paint getBorderPaint();

    public abstract int getRunnerBorderColor();
    public abstract Paint getRunnerBorderPaint();

    public abstract int getBorderColor();

    public abstract Paint getScaleBorderPaint();

    public abstract int[] getDegreeColors();

    public abstract int getBaseFretColor(int fretNumber);

    public abstract int getFretboardRunnerPaintColor(int fretNumber);

    //Got replaced by a maximal color difference algorithm... may come back
    public abstract int getFretboardRunnerTextColor(int fretNumber);

    public abstract float getBorderStrokeWidth();

    public abstract int[] getIntervalColors();

    public abstract int getEmptyFretColor();

    public abstract int getFretboardBorderColor();

    public abstract int getTextStrokeColor();

}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/themes/ShredsheetsTheme.kt
package com.bm.shredsheets.models.themes

import com.badlogic.gdx.graphics.Color
import com.bm.shredsheets.models.Paint

abstract class ShredsheetsTheme {
    var defaultVector =
        booleanArrayOf(true, false, true, false, true, false, true, false, false, false, false, false)
    abstract var degreeHighlightingVector: BooleanArray?

    abstract val borderPaint: Paint?
    abstract val runnerBorderColor: Color
    abstract val runnerBorderPaint: Paint?
    abstract val borderColor: Color
    abstract val scaleBorderPaint: Paint?
    abstract val degreeColors: Array<Color>

    abstract fun getBaseFretColor(fretNumber: Int): Color
    abstract fun getFretboardRunnerPaintColor(fretNumber: Int): Color

    //Got replaced by a maximal color difference algorithm... may come back
    abstract fun getFretboardRunnerTextColor(fretNumber: Int): Color
    abstract val borderStrokeWidth: Float
    abstract val intervalColors: Array<Color>
    abstract val emptyFretColor: Color
    abstract val fretboardBorderColor: Color
    abstract val textStrokeColor: Color

    companion object {
        val themes: Array<ShredsheetsTheme>
            get() = arrayOf(DefaultTheme())
    }
}
